include/leaderboard.hpp

```diff
@@ -0,0 +1,78 @@
+#ifndef PLANETGAME_LEADERBOARD_HPP
+#define PLANETGAME_LEADERBOARD_HPP
+
+#include <map>
+#include <string>
+#include <fstream>
+#include <sstream>
+#include <iostream>
+
+#include <SFML/Graphics.hpp>
+
+#include "textures.hpp"
+
+/**
+ * Leaderboard class that manages the scores of the player and displays them on the window.
+ */
+class Leaderboard {
+public:
+    using ScoreMap = std::multimap<unsigned int, std::string, std::greater<>>;
+
+    /**
+     * Constructs the leaderboard.
+     */
+    Leaderboard() = default;
+
+    /**
+     * Gets scores with corresponding player names from file.
+     */
+    void retrieveScores();
+
+    /**
+     * Adds a new score to the leaderboard.
+     *
+     * @param score the score to add
+     * @param name the name of the player
+     */
+    void addScore(unsigned int score, std::string name = "Student");
+
+    /**
+     * Writes scores back to file.
+     */
+    void saveScores();
+
+    /**
+     * Returns the scores.
+     *
+     * @return the scores
+     */
+    [[nodiscard]] ScoreMap getScores() const {
+        return scores;
+    }
+
+    /**
+     * Calls three functions: retrieveScores, addScore, and saveScores.
+     *
+     * @param score the score to update
+     */
+    void updateScore(unsigned int score);
```

Comment:
I've decided to create a single function for calling other three, since it seems logical to me to have one dedicated function for updating score in the `Leaderboard` instance (so we can just call `leaderboard.updateScore(playerScore, playerName);` when we got both arguments and don't think of how it would be updated under the hood), and at the same time not to merge all logic into one function, since we are teaching OOP and delegating one function for one task is a good practice.